applications/FSIApplication/custom_utilities/ibqn_mvqn_convergence_accelerator.h

```diff
@@ -0,0 +1,455 @@
+     */
+    void InitializeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        mConvergenceAcceleratorIteration = 0;
+
+        mpConvergenceAcceleratorLeft->InitializeSolutionStep();
+        mpConvergenceAcceleratorRight->InitializeSolutionStep();
+
+        KRATOS_CATCH( "" );
+    }
+
+    void InitializeNonLinearIteration() override
+    {
+        KRATOS_TRY;
+
+        mpConvergenceAcceleratorLeft->InitializeNonLinearIteration();
+        mpConvergenceAcceleratorRight->InitializeNonLinearIteration();
+
+        KRATOS_CATCH( "" );
+    }
+
+    /**
+     * @brief Performs the solution update
+     * This method computes the solution update using a Jacobian approximation.
+     * Such Jacobian approximation is computed with the MVQN (MultiVector Quasi-Newton method) algorithm.
+     * @param rResidualVector Residual vector from the residual evaluation
+     * @param rIterationGuess Current iteration guess to be corrected. Should be initialized to zero outside the convergence accelerator.
+     */
+    void UpdateSolution(
+        const VectorType& rResidualVector,
+        VectorType& rIterationGuess) override
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR << "The UpdateSolution() method cannot be called in the interface block Newton case. Use either \'UpdateSolutionLeft\' or \'UpdateSolutionRight\'" << std::endl;
+
+        KRATOS_CATCH( "" );
+    }
+
+    //TODO: CREATE AN INTERMEDIATE IBQN BASE CLASS
+    virtual void UpdateSolutionRight(
+        const VectorType& rForceInputVector,
+        const VectorType& rDisplacementOutputVector,
+        VectorType& rIterationGuess)
+    {
+        // Save the uncorrected force vector for the displacement update
+        auto p_uncor_disp_aux = Kratos::make_shared<VectorType>(rDisplacementOutputVector);
+        std::swap(mpUncorrectedDisplacementVector, p_uncor_disp_aux);
+
+        // Update right inverse Jacobian approximation with the current information
+        mpConvergenceAcceleratorRight->UpdateInverseJacobianApproximation(rForceInputVector, rDisplacementOutputVector);
+
+        VectorType right_correction(TSparseSpace::Size(rIterationGuess));
+        if (!mFirstRightCorrectionPerformed) {
+            // Calculate the first correction as a fixed point relaxation
+            TSparseSpace::SetToZero(right_correction);
+            TSparseSpace::UnaliasedAdd(right_correction, mInitialOmega, rDisplacementOutputVector - rIterationGuess);
+
+            // Update the first iteration correction flag
+            mFirstRightCorrectionPerformed = true;
+
+        } else {
+            // Get the interface problem size
+            std::size_t problem_size = mpConvergenceAcceleratorRight->GetProblemSize();
+
+            // Get both left and inverse Jacobian
+            auto p_inv_jac_right = mpConvergenceAcceleratorRight->pGetInverseJacobianApproximation();
+            auto p_inv_jac_left = mpConvergenceAcceleratorLeft->pGetInverseJacobianApproximation();
```

Comment:
I am thinking if this should also be in the `public` Interface of the Baseclass
this way you wouldn't need the `friend` declarations

Are you planning on adding more MVQN types for this Acc?
If no then maybe the `friend` is enough an we can take out the confusing protected interface (as it is not needed due to `friend`

Reply:
As it is right now, the MVQN will only work with this one. Depending on how the design evolves (see my comment above) we could (ideally) have a unique `IBQN` class that works with several Jacobian approximators such as the IBQN-ILS (the point is that the interface equations are the same). Then we will need for sure to remove the `friend`.